### PR TITLE
Add popup menu support for chat header and chat list

### LINKED ISSUE #5311: Implement chat secondary actions on left panel
**Story**
As a user, I want to be able to find secondary actions on 1:1 and public chats when I right-click on the user's row on the left panel.

**Actual behaivor**
Nothing happens when right clicking on a 1:1 or public chat row on left panel.

**Expected behavior**
When right clicking on "Super Violet Coordinated" row:
captura de pantalla 2018-07-26 a la s 4 02 01 p m
When right clicking on "#peoplewholovedogs" row:
captura de pantalla 2018-07-26 a la s 4 06 20 p m
Note: This behavior was observed on most of participants during [usability test](https://docs.google.com/document/d/1Ny4QRw7zLSmH5rO8ZhwU313BlbWhGGwOj4FTIPwh2tQ/edit?pli=1).
Designs (Zeplin) >> zpl.io/V4J4NX4

### PR DESCRIPTION
**Pull request by vitvly**
Fixes #5311 #5314

### CODE REVIEW

File: components/src/status_im/ui/components/react.cljs [hunk 1]
```diff
@@ -68,6 +68,12 @@
 (def activity-indicator (get-class "ActivityIndicator"))
 
 (def modal (get-class "Modal"))
+(def popup-menu-provider (adapt-class (.-MenuProvider js-dependencies/popup-menu)))
+(def popup-menu (adapt-class (.-Menu js-dependencies/popup-menu)))
+(def popup-menu-options (adapt-class (.-MenuOptions js-dependencies/popup-menu)))
+(def popup-menu-option (adapt-class (.-MenuOption js-dependencies/popup-menu)))
+(def popup-menu-trigger (adapt-class (.-MenuTrigger js-dependencies/popup-menu)))
+(def popup-menu-renderers (js->clj (.-renderers js-dependencies/popup-menu) :keywordize-keys true))
```
Comment (rcullito) [inline components/src/status_im/ui/components/react.cljs]: I'm lacking context here clearly, but what is the guideline for converting to clj data structure in the last form versus `adapt-class` in the others? Thank you!

File: src/status_im/ui/screens/desktop/main/tabs/home/views.cljs [hunk 2]
```diff
@@ -62,9 +63,26 @@
...
         [chat-item/message-timestamp (:timestamp last-message)]]])))
 
-(defn chat-list-item [[chat-id chat]]
-  [react/touchable-highlight {:on-press #(re-frame/dispatch [:chat.ui/navigate-to-chat chat-id])}
-   [chat-list-item-inner-view (assoc chat :chat-id chat-id)]])
+(defn chat-list-item [[chat-id
+                       {:keys [group-chat public? public-key] :as chat}]]
+  [popup-menu/wrap-with-menu
+   [chat-list-item-inner-view (assoc chat :chat-id chat-id)]
+   (remove nil?
```
Comment (rcullito) [inline src/status_im/ui/screens/desktop/main/tabs/home/views.cljs]: which set of chats are expected to not be removed here? just private, or are we up to additional categories? :eye:

**Review by rcullito — APPROVED** [review]
Nice PR! Included a couple questions inline.

File: components/src/status_im/ui/components/react.cljs (hunk 1 above)
Comment (vitvly) [reply components/src/status_im/ui/components/react.cljs]: `renderers` is not a class, but a folder inside `react-native-popup-menu`. In this case `(.-renderers js-dependencies/popup-menu)` returns a js object with string keys (map of class names to objects) So we convert it to a Clojure-style keyword map.

File: src/status_im/ui/screens/desktop/main/tabs/home/views.cljs (hunk 2 above)
Comment (vitvly) [reply src/status_im/ui/screens/desktop/main/tabs/home/views.cljs]: `remove` affects menu items for this particular chat item, not the chat list. So for public/group chats we cannot execute `View Profile`, for example.  Will write the doc for `wrap-with-menu`, sorry for confusion.

File: components/src/status_im/ui/components/react.cljs (hunk 1 above)
Comment (rcullito) [reply components/src/status_im/ui/components/react.cljs]: great, thanks for clarifying.

File: src/status_im/ui/screens/desktop/main/tabs/home/views.cljs (hunk 2 above)
Comment (rcullito) [reply src/status_im/ui/screens/desktop/main/tabs/home/views.cljs]: ah, no that makes total sense. my own confusion for missing at what level the `remove` was acting upon. thanks!

**Review by maxhora — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

**Comment by churik** [comment]
@siphiuel can it be tested? If yes, can you please rebase it and move to `TO TEST ` column? Thanks!

**Comment by vitvly** [comment]
Rebased and moved to `TO TEST`.

**Comment by churik** [comment]
@SlizkovaGanna 
any blockers here so far? do you need any help?

**Comment by SlizkovaGanna** [comment]
Tested on MacOS and Linux (emulated)

Issues found:

**1. The 'Unknown' user it triggered when selecting 'View profile' on right-click from the chat list, _reproduces for all contacts every time_**
**Steps to reproduce:**
1. Login to Status and go to your Home screen with at least 1-1 chat with a contact
2. Right-click on that chat and select 'View profile' from the menu 
**Expected result:** User's profile is shown with 'Add to contacts' being inactive
**Current result:** 'Unknown' profile is displayed, not from your contacts.
Note: If you add that 'Unknown' to contacts - you will see two people in contact list - the initial contact you had 1-1 chat with at the start, and the 'Unknown'

image

3. Click 'Send message' - a new 1-1 chat with the 'Unknown' will appear
**Expected result:** User is redirected to the already existing 1-1 chat
**Current result:** A new 1-1 chat with the 'Unknown' will appear

image

4. Start typing a message
**Expected result:** Message input as usual
**Current result:** With every symbol, 'Unknown''s icon changes to a different default one in the chat list on the left

image
image

**2. When triggering the right-click menu in the chat list, the menu contains options for the chat that the mouse pointer is at, but performs actions for the chat that is selected at the moment**
In some cases, it leads to even crashing the app:

image

**Steps to reproduce:**
1. Create a group chat, 1-1 chat and join a public chat

image

2. Select the 1-1 chat, but right-click the group chat area and select 'Group info'
**Expected result:** Not sure what the expected result it - a menu for the highlighted chat (if so - there should be no 'group info' option) or the chat that the mouse pointer is at (if so - group info of the group chat should be displayed); either way menus should not mix up like they do
**Current result:** The group info of 1-1 chat is displayed, as if it was a group chat:

image

Same for a public chat:

image

**Also tested:**
- view profile/group info (1-1, group), both menus 
- deleting history (1-1, group, public) and checking it doesn't re-appear, both menus
- deleting chats (1-1, group, public) and checking it doesn't re-appear, both menus
- messaging (regular, different symbols, offline, fetching messages)

**Comment by churik** [comment]
1. Known issue https://github.com/status-im/status-react/issues/6210

**Comment by asemiankevich** [comment]
@rcullito @MaxRis builds are failing in Jenkins, i restarted it not sure about success, please take a look https://ci.status.im/job/status-react/job/pull%20requests/view/change-requests/job/PR-6671/

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/pull%20requests/view/change-requests/job/PR-6671/">PR-6671 [status-react » pull requests] [Jenkins]</a></strong></div></blockquote>

**Comment by asemiankevich** [comment]
@siphiuel please rebase :)

**Comment by SlizkovaGanna** [comment]
Tested on MacOS and Linux (emulated)

There were no specific PR-related issues found, all the general ones found will be reported separately.
The only things I can possible name is a different design of the pop-up window on Linux

Linux:
image

MaxOS:
image

@siphiuel if its possible to fix, I'll create a separate low priority issue for it.

**Tested:**
- The 'Unknown' user it triggered when selecting 'View profile' on right-click from the chat list, reproduces for all contacts every time - **fixed**

@Serhy Also checked scenarios for 'Unknown' issue from https://github.com/status-im/status-react/issues/6210
scenario #1 - waiting for 24 h to pass
scenario #2 - does not reproduce on this build
Would need to recheck the general issue after this PR is merged.

- When triggering the right-click menu in the chat list, the menu contains options for the chat that the mouse pointer is at, but performs actions for the chat that is selected at the moment - **fixed**

- view profile/group info (1-1, group), both menus
- deleting history (1-1, group, public) and checking it doesn't re-appear, both menus
- deleting chats (1-1, group, public) and checking it doesn't re-appear, both menus
- messaging (regular, different symbols, offline, fetching messages)
- pairing

Automated tests fails are expected.
Tested on MacOs and Linux (emulated)

**Pull request merged**